content/en/blog/_posts/2020-03-30-two-phased-canary-rollout-with-gloo.md

```diff
@@ -0,0 +1,772 @@
+            
+```bash
+kubectl delete deploy -n echo echo-v1
+```
+
+Now our cluster looks like this:
+
+![](https://raw.githubusercontent.com/solo-io/gloo-ref-arch/blog-30-mar-20/platform/prog-delivery/two-phased-with-os-gloo/4-decommissioning-v1/end-state.png)
+
+And requests to the gateway return this:
+
+```bash
+➜ curl $(glooctl proxy url)/
+version:v2
+```
+
+We have now completed our two-phased canary rollout of an application update using Gloo!
+
+### Checking our work
+
+Envoy publishes a large number of stats that we can scrape and use to track the health of downstream requests through
+the proxy. 
+
+Running through these steps over a 15 minute period while sending a significant load to these routes on the proxy 
+yielded the following operational dashboards:
+
+![](https://raw.githubusercontent.com/solo-io/gloo-ref-arch/blog-30-mar-20/platform/prog-delivery/two-phased-with-os-gloo/perf-test.png)
+
+The charts show a fluctuating load of 90-250 rps hitting the proxy. While the two phased rollout was commencing, we can 
+observe that Envoy experienced no spikes in request latency, nor did it respond with any errors. 
+
+## Other Advanced Topics
+
+Over the course of this post, we collected a few topics that could be a good starting point for advanced exploration:
+
+* Using the **JWT** filter to verify JWTs, extract claims onto headers, and route to canary versions depending on a claim value. 
+* Looking at **Prometheus metrics** and **Grafana dashboards** created by Gloo to monitor the health of the rollout. 
+* Automating the rollout by integrating **Flagger** with **Gloo**
+
+A few other topics that warrant further exploration:
+
+* Supporting **self-service** upgrades by giving teams ownership over their upstream and route configuration
+* Utilizing Gloo's **delegation** feature and Kubernetes **RBAC** to decentralize the configuration management safely
+* Fully automating the continuous delivery process by applying **GitOps** principles and using tools like **Flux** to push config to the cluster
+* Supporting **hybrid** or **non-Kubernetes** application use-cases by setting up Gloo with a different deployment pattern
+* Utilizing **traffic shadowing** to begin testing the new version with realistic data before shifting production traffic to it
+
+## Get Involved in the Gloo Community
+
+Gloo has a large and growing community of open source users, in addition to an enterprise customer base. To learn more about 
+Gloo:
+
+* Check out the [repo](https://github.com/solo-io/gloo), where you can see the code and file issues
+* Check out the [docs](https://docs.solo.io/gloo/latest), which have an extensive collection of guides and examples
+* Join the [slack channel](http://slack.solo.io/) and start chatting with the Solo engineering team and user community
+
+If you'd like to get in touch with me (feedback is always appreciated!), you can find me on Slack or email me at 
+**rick.ducott@solo.io**. 
```

Comment:
Which Slack? Solo Slack or Kubernetes Slack? Or both?

Reply:
Thanks - technically either but I linked to solo slack :D